coveo-settings/coveo_settings/settings.py

```diff
@@ -166,20 +203,26 @@ def _get_value(self) -> Optional[ConfigValue]:
         if value is None and self._fallback is not None:
             value = self._fallback() if callable(self._fallback) else self._fallback
 
+        self._last_value = copy(value)
         return value
 
     def _raise_if_missing(self) -> None:
         """ Raises an MandatoryConfigurationError exception if the setting is required but missing. """
         if not self.is_set:
             raise MandatoryConfigurationError(f'Mandatory config item "{self.key}" is missing.')
 
-    def _pretty_repr(self, value: Optional[ConfigValue]) -> str:
+    def _pretty_repr(self, value: Optional[Any]) -> str:
         value_str = "<sensitive>" if self._sensitive else "<not-set>" if value is None else value
         return f"{self.__class__.__name__}[{self.key}] = {value_str}"
 
     def __repr__(self) -> str:
         """ Returns a readable representation of the item for debugging. """
-        return self._pretty_repr(self._get_value())
+        # we are overly careful in not triggering mechanics (e.g.: _get_value()) from here.
+        # value is only shown if already computed.
+        value = next(
+            _ for _ in (self._cache_validated, self._last_value, "<not-evaluated>") if _ is not None
```

Comment:
[detail] I'm used to see `_` for variables that are ignore. Here, the variable is not ignored.

Reply:
i love saving characters but these days you get judged by using `i` so now I use `_` 😆